kube-hunter.py

```diff
@@ -1,63 +1,55 @@
 #!/usr/bin/env python3
+
 import argparse
 import logging
 import threading
 
+from src.conf import config
+from src.modules.report.plain import PlainReporter
+from src.modules.report.yaml import YAMLReporter
+from src.modules.report.json import JSONReporter
+from src.modules.report.dispatchers import STDOUTDispatcher, HTTPDispatcher
+from src.core.events import handler
+from src.core.events.types import HuntFinished, HuntStarted
+from src.modules.discovery.hosts import RunningAsPodEvent, HostScanEvent
+
 
-parser = argparse.ArgumentParser(description='Kube-Hunter - hunts for security weaknesses in Kubernetes clusters')
-parser.add_argument('--list', action="store_true", help="displays all tests in kubehunter (add --active flag to see active tests)")
-parser.add_argument('--interface', action="store_true", help="set hunting of all network interfaces")
-parser.add_argument('--pod', action="store_true", help="set hunter as an insider pod")
-parser.add_argument('--quick', action="store_true", help="Prefer quick scan (subnet 24)")
-parser.add_argument('--include-patched-versions', action="store_true", help="Don't skip patched versions when scanning")
-parser.add_argument('--cidr', type=str, help="set an ip range to scan, example: 192.168.0.0/16")
-parser.add_argument('--mapping', action="store_true", help="outputs only a mapping of the cluster's nodes")
-parser.add_argument('--remote', nargs='+', metavar="HOST", default=list(), help="one or more remote ip/dns to hunt")
-parser.add_argument('--active', action="store_true", help="enables active hunting")
-parser.add_argument('--log', type=str, metavar="LOGLEVEL", default='INFO', help="set log level, options are: debug, info, warn, none")
-parser.add_argument('--report', type=str, default='plain', help="set report type, options are: plain, yaml, json")
-parser.add_argument('--dispatch', type=str, default='stdout', help="where to send the report to, options are: stdout, http (set KUBEHUNTER_HTTP_DISPATCH_URL and KUBEHUNTER_HTTP_DISPATCH_METHOD environment variables to configure)")
-parser.add_argument('--statistics', action="store_true", help="set hunting statistics")
-
-import plugins
-
-config = parser.parse_args()
-
-try:
-    loglevel = getattr(logging, config.log.upper())
-except:
-    pass
+# TODO: move log level parsing to conf module
```

Comment:
should we create a new issue to track the conf refactor and add these TODOs there? I understand you don't want to block this PR on all other little refactoring but we should be able to better track it

Reply:
I believe it is a good idea

Reply:
created https://github.com/aquasecurity/kube-hunter/issues/289. please remove the comments before merging